website/docs/recipes/kafka.md

```diff
@@ -4,4 +4,168 @@ title: Kafka
 ---
 
-TBC
+The code examples here use Java, and are based on writing JSON messages to Kafka. If you are using AVRO or Protobuf, then the Kafka Schema-Registry examples may not yet be implemented.
+
+## Kafka JSON Consumer
+We'll start with the bones of a class and we will use Junit5 annotations:
+
+```
+@ExtendWith(PactConsumerTestExt.class)
+@PactTestFor(providerName = "jsonKafkaProviderApp", providerType = ProviderType.ASYNCH, pactVersion = PactSpecVersion.V3)
+class JsonKafkaConsumerTest {
+...
+}
+```
+
+Although we're writing the consumer test, we've also named the provider. Next, we'll define what we *expect* the contract to look like:
+```
+@Pact(consumer = "jsonKafkaConsumerApp")
+MessagePact simpleJsonPact(MessagePactBuilder builder) {
+    PactDslJsonBody body = new PactDslJsonBody();
+    body.stringType("name", "almost-anything");
+
+    return builder.expectsToReceive("A simple message")
+                  .withMetadata(Map.of("contentType", "application/json"))
+                  .withContent(body)
+                  .toPact();
+}
+```
+The most interesting part here is the body. We've defined it to have a single String field - I'd expect your real one to have more than one field in all likelihood. Having defined what we're expecting, we can finally write our test:
+```
+@Test
+@PactTestFor(pactMethod = "simpleJsonPact", providerType = ProviderType.ASYNCH)
+void simpleMessage(List<Message> messages) {
+    byte[] kafkaBytes = convertToKafkaBytes(messages.get(0));
+
+    assertDoesNotThrow(() -> {
+        expectApplicationToConsumeKafkaBytesSuccessfully(kafkaBytes);
+    });
+}
+
+private byte[] convertToKafkaBytes(Message message) {
+    return message.contentsAsBytes();
+}
+
+private void expectApplicationToConsumeKafkaBytesSuccessfully(byte[] kafkaBytes) {
+    ConsumerDomainRecord consumerDomainRecord = useProductionCodeToDeserializeKafkaBytesToDomain(kafkaBytes);
+
+    ProductionCode productionCode = new ProductionCode();
+    productionCode.handle(consumerDomainRecord);
+}
+
+private ConsumerDomainRecord useProductionCodeToDeserializeKafkaBytesToDomain(byte[] kafkaBytes) {
+    Deserializer<ConsumerDomainRecord> deserializer = getProductionKafkaDeserializer();
+    return deserializer.deserialize("", kafkaBytes);
+}
+
+private Deserializer<ConsumerDomainRecord> getProductionKafkaDeserializer() {
+    KafkaJsonDeserializer<ConsumerDomainRecord> domainRecordKafkaJsonDeserializer = new KafkaJsonDeserializer<>();
+    Map<String, Object> props = Map.of(
+            KafkaJsonDeserializerConfig.JSON_VALUE_TYPE, ConsumerDomainRecord.class.getName()
+    );
+    domainRecordKafkaJsonDeserializer.configure(props, false);
+    return domainRecordKafkaJsonDeserializer;
+}
+```
+
+Our test method gets a List of Messages passed in - these are PACT's representation of the message(s) that we defined in the previous step. Each message has space for a byte array and also metadata. We're not going to look much at metadata at this point. 
```

Comment:
`s/PACT/Pact`

Reply:
Will do... Thanks.
Didn't realise anyone was already looking at it... I got some complaints about the commit message, just tried a force push.